[BUG]Nil pointer dereference when namespace churning with cluster-scoped resources

# Bug Description

When running a job with churnConfig.mode: namespaces that contains cluster-scoped (non-namespaced) resources, kube-burner panics with a nil pointer dereference during the namespace re-creation phase.

## **Output of `kube-burner` version**
  - kube-burner version: v2.1.0
  - Go version: 1.23.12
  - 
## **Describe the bug**
 https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_release/72959/rehearse-72959-periodic-ci-openshift-eng-ocp-qe-perfscale-ci-main-gcp-4.20-nightly-x86-olmv1-benchmark-test/2004389471251861504/artifacts/olmv1-benchmark-test/olmv1-performance/build-log.txt
```go
time="2025-12-26 04:46:14" level=info msg="Re-creating 1 deleted namespaces" file="create.go:449"
time="2025-12-26 04:46:14" level=debug msg="Created namespace: olmv1-ce" file="namespaces.go:53"
time="2025-12-26 04:46:14" level=info msg="0/1 iterations completed" file="create.go:134"
time="2025-12-26 04:46:14" level=debug msg="Creating object replicas from iteration 0" file="create.go:140"
E1226 04:46:14.692899     165 panic.go:262] "Observed a panic" panic="runtime error: invalid memory address or nil pointer dereference" panicGoValue="\"invalid memory address or nil pointer dereference\"" stacktrace=<
	goroutine 5436 [running]:
	k8s.io/apimachinery/pkg/util/runtime.logPanic({0x2f664c8, 0x48d1ee0}, {0x26d5080, 0x487b690})
		/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.31.1/pkg/util/runtime/runtime.go:107 +0xbc
	k8s.io/apimachinery/pkg/util/runtime.handleCrash({0x2f664c8, 0x48d1ee0}, {0x26d5080, 0x487b690}, {0x48d1ee0, 0x0, 0x437465?})
		/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.31.1/pkg/util/runtime/runtime.go:82 +0x5e
	k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc0014021c0?})
		/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.31.1/pkg/util/runtime/runtime.go:59 +0x108
	panic({0x26d5080?, 0x487b690?})
		/opt/hostedtoolcache/go/1.23.12/x64/src/runtime/panic.go:791 +0x132
	k8s.io/apimachinery/pkg/apis/meta/v1/unstructured.(*Unstructured).GetKind(...)
		/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.31.1/pkg/apis/meta/v1/unstructured/unstructured.go:231
	github.com/kube-burner/kube-burner/v2/pkg/burner.(*JobExecutor).createRequest.func1()
		/home/runner/go/pkg/mod/github.com/kube-burner/kube-burner/v2@v2.1.0/pkg/burner/create.go:352 +0x635
	k8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtection(0xc000f945e8?)
		/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.31.1/pkg/util/wait/wait.go:145 +0x3e
	k8s.io/apimachinery/pkg/util/wait.ExponentialBackoff({0x3b9aca00, 0x4008000000000000, 0x0, 0x9, 0x0}, 0xc001e8ae70)
		/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.31.1/pkg/util/wait/backoff.go:461 +0x5a
	github.com/kube-burner/kube-burner/v2/pkg/util.RetryWithExponentialBackOff(0xc000f94670, 0x3b9aca00, 0x4008000000000000, 0x0, 0x2?)
		/home/runner/go/pkg/mod/github.com/kube-burner/kube-burner/v2@v2.1.0/pkg/util/utils.go:41 +0xd6
	github.com/kube-burner/kube-burner/v2/pkg/burner.(*JobExecutor).createRequest(0xc0011e0e00?, {0x2f66650?, 0xc0001fee70?}, {{0xc000c122d0, 0x18}, {0xc000762e6c, 0x2}, {0xc000762e70, 0xf}}, {0x0, ...}, ...)
		/home/runner/go/pkg/mod/github.com/kube-burner/kube-burner/v2@v2.1.0/pkg/burner/create.go:310 +0x1d4
	github.com/kube-burner/kube-burner/v2/pkg/burner.(*JobExecutor).replicaHandler.func1.1({0xc000383ba8?, 0x1?})
		/home/runner/go/pkg/mod/github.com/kube-burner/kube-burner/v2@v2.1.0/pkg/burner/create.go:251 +0x129
	created by github.com/kube-burner/kube-burner/v2/pkg/burner.(*JobExecutor).replicaHandler.func1 in goroutine 5435
		/home/runner/go/pkg/mod/github.com/kube-burner/kube-burner/v2@v2.1.0/pkg/burner/create.go:246 +0x616
 >
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x22d4975]

goroutine 5436 [running]:
k8s.io/apimachinery/pkg/util/runtime.handleCrash({0x2f664c8, 0x48d1ee0}, {0x26d5080, 0x487b690}, {0x48d1ee0, 0x0, 0x437465?})
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.31.1/pkg/util/runtime/runtime.go:89 +0xee
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc0014021c0?})
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.31.1/pkg/util/runtime/runtime.go:59 +0x108
panic({0x26d5080?, 0x487b690?})
	/opt/hostedtoolcache/go/1.23.12/x64/src/runtime/panic.go:791 +0x132
k8s.io/apimachinery/pkg/apis/meta/v1/unstructured.(*Unstructured).GetKind(...)
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.31.1/pkg/apis/meta/v1/unstructured/unstructured.go:231
github.com/kube-burner/kube-burner/v2/pkg/burner.(*JobExecutor).createRequest.func1()
	/home/runner/go/pkg/mod/github.com/kube-burner/kube-burner/v2@v2.1.0/pkg/burner/create.go:352 +0x635
k8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtection(0xc000f945e8?)
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.31.1/pkg/util/wait/wait.go:145 +0x3e
k8s.io/apimachinery/pkg/util/wait.ExponentialBackoff({0x3b9aca00, 0x4008000000000000, 0x0, 0x9, 0x0}, 0xc000f0de70)
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.31.1/pkg/util/wait/backoff.go:461 +0x5a
github.com/kube-burner/kube-burner/v2/pkg/util.RetryWithExponentialBackOff(0xc000f94670, 0x3b9aca00, 0x4008000000000000, 0x0, 0x2?)
	/home/runner/go/pkg/mod/github.com/kube-burner/kube-burner/v2@v2.1.0/pkg/util/utils.go:41 +0xd6
github.com/kube-burner/kube-burner/v2/pkg/burner.(*JobExecutor).createRequest(0xc0011e0e00?, {0x2f66650?, 0xc0001fee70?}, {{0xc000c122d0, 0x18}, {0xc000762e6c, 0x2}, {0xc000762e70, 0xf}}, {0x0, ...}, ...)
	/home/runner/go/pkg/mod/github.com/kube-burner/kube-burner/v2@v2.1.0/pkg/burner/create.go:310 +0x1d4
github.com/kube-burner/kube-burner/v2/pkg/burner.(*JobExecutor).replicaHandler.func1.1({0xc000383ba8?, 0x1?})
	/home/runner/go/pkg/mod/github.com/kube-burner/kube-burner/v2@v2.1.0/pkg/burner/create.go:251 +0x129
created by github.com/kube-burner/kube-burner/v2/pkg/burner.(*JobExecutor).replicaHandler.func1 in goroutine 5435
	/home/runner/go/pkg/mod/github.com/kube-burner/kube-burner/v2@v2.1.0/pkg/burner/create.go:246 +0x616
+ exit_code=2
++ date -u +%Y-%m-%dT%H:%M:%SZ
+ JOB_END=2025-12-26T04:46:14Z
```
A clear and concise description of what the bug is.

## **To Reproduce**

  1. Create a job configuration with:
    - churnConfig.mode: namespaces (or use --churn-mode namespaces)
    - churnConfig.cycles: 1 (or greater)
    - Objects that include cluster-scoped resources (e.g., ClusterRole, ClusterRoleBinding, ClusterCatalog, ClusterExtension)
  2. Run kube-burner with this configuration
  3. Wait for the churn cycle to delete and re-create namespaces

## **Expected behavior**

A clear and concise description of what you expected to happen.

 Cluster-scoped resources should be skipped during namespace churning (since they weren't deleted with the namespace), and the job should continue without error.

## **Screenshots or output**

If applicable, add screenshots or kube-burner output to help explain your problem.

## **Additional context**

Add any other context about the problem here.

--- a/pkg/burner/create.go
+++ b/pkg/burner/create.go
@@ -320,6 +320,10 @@ func (ex *JobExecutor) createRequest(ctx context.Context, gvr schema.GroupVersio
 		} else {
 			if !ex.nsChurning {
 				uns, err = ex.dynamicClient.Resource(gvr).Create(context.TODO(), obj, metav1.CreateOptions{})
+			} else {
+				// Skip non-namespaced objects during namespace churning - they won't be deleted with the namespace
+				log.Debugf("Skipping non-namespaced object %s/%s during namespace churning", obj.GetKind(), obj.GetName())
+				return true, nil
 			}
 		}
 		if err != nil {

